crates/bevy_time/src/delayed_commands.rs

```diff
@@ -0,0 +1,79 @@
+use crate::Time;
+use alloc::boxed::Box;
+use bevy_ecs::prelude::*;
+use bevy_log::warn;
+use core::time::Duration;
+
+/// A [`Command`] that will be executed after a specified delay has elapsed.
+///
+/// This can be helpful for scheduling actions at some point in the future.
+///
+/// This works by moving the supplied command into a component that is spawned on an entity.
+/// Delayed command entities are ticked via [`tick_delayed_commands`],
+/// which is typically run in [`First`] as part of [`TimePlugin`].
+#[derive(Component)]
+pub struct DelayedCommand {
+    pub delay: Duration,
+    pub command: Box<dyn Command + Send + Sync + 'static>,
+}
+
+impl DelayedCommand {
+    pub fn new(delay: Duration, command: impl Command + Send + Sync + 'static) -> Self {
+        Self {
+            delay,
+            command: Box::new(command),
+        }
+    }
+}
+
+impl Command for DelayedCommand {
+    /// Spawns a new entity with the [`DelayedCommand`] as a component.
+    fn apply(self, world: &mut World) {
+        world.spawn(self);
+    }
+}
```

Comment:
This feels weird to me. Why not have users spawn/insert it like other components? We don't `impl Command for Observer`, for example.

Reply:
Oh, that's an interesting API idea 🤔